src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/pre_compute.cpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "../../logging.hpp"
+#include "pre_compute.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/pass/pattern/op/optional.hpp"
+#include "openvino/op/ops.hpp"
+#include "ov_ops/rotary_positional_embeddings.hpp"
+#include "transformations/common_optimizations/fuse_rotary_positional_embeddings.hpp"
+#include "openvino/pass/manager.hpp"
+
+namespace opp = ov::pass::pattern;
+
+namespace {
+    uint32_t align_to(uint32_t value, uint32_t alignment) {
+        return (value + alignment - 1) & ~(alignment - 1);
+    }
+    template <typename T>
+    std::shared_ptr<ov::Node> makeConst(const ov::element::Type& type, const ov::Shape& shape, const std::vector<T>& values) {
+        return std::make_shared<ov::op::v0::Constant>(type, shape, values);
+    }
+    //TODO: copied from common tests
+    static ov::OutputVector makeCosSinCache(size_t max_position_embeddings, std::shared_ptr<ov::Node> inverse_frequencies) {
```

Comment:
probably vector isn't the right type here, maybe a pair suits better?

Reply:
vector is fine - standard return type for ov output nodes collection